Sources/InfomaniakCore/Background/ExpiringActivity.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+/// Delegation mechanism to notify the end of an `ExpiringActivity`
+public protocol ExpiringActivityDelegate: AnyObject {
+    /// Called when the system is requiring us to terminate an expiring activity
+    func backgroundActivityExpiring()
+}
+
+/// Something that can perform arbitrary short background tasks, with a super simple API.
+public protocol ExpiringActivityable {
+    /// Common init method
+    /// - Parameters:
+    ///   - id: Something to identify the background activity in debug
+    ///   - qos: QoS used by the underlying queues
+    ///   - delegate: The delegate to notify we should terminate
+    init(id: String, qos: DispatchQoS, delegate: ExpiringActivityDelegate?)
+
+    /// init method
+    /// - Parameters:
+    ///   - id: Something to identify the background activity in debug
+    ///   - delegate: The delegate to notify we should terminate
+    init(id: String, delegate: ExpiringActivityDelegate?)
+
+    /// Register with the system an expiring activity
+    func start()
+
+    /// Terminate all the expiring activities
+    func endAll()
+}
+
+public final class ExpiringActivity: ExpiringActivityable {
```

Comment:
This is something in `Drive Core` that I brought here, also cleaning it a bit, adding some tests, documenting the protocol. The whole 9 yards.